### PR TITLE
Adding examples!

### LINKED ISSUE #575: Add examples directory with some common Dust implementation examples
This is a common pattern for libraries on GitHub.

* Each example should have an explanation page on dustjs.com/examples/ that links to the example.
* Each example should have a package.json or bower.json to allow it to install dependencies and Just Work™.

Some good examples could be:

* Basic Node example
* Basic Node with Express example
* Basic streaming example
* Basic browser example
* Loading Dust via AMD

### PR DESCRIPTION
**Pull request by sethkinast**
Simple browser
AMD in the browser
Super-simple Node
Streaming with Express
Streaming with Hoffman

Closes #575

### CODE REVIEW

**Comment by smfoote** [comment]
These examples are awesome! Nice work Seth. Sorry you missed your bus.

File: examples/basic-node/app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var dust = require('dustjs-linkedin');
+
+var tmpl = dust.compile("Hello world! Using Dust version {version}!", "hello");
+dust.loadSource(tmpl);
```
Comment (jimmyhchan) [inline examples/basic-node/app.js]: can we have the example pre-compile the templates. not a fan of `loadsource`.

Comment (sethkinast) [inline examples/basic-node/app.js]: The other examples do pre-compile, but this was intended to be the most simple example possible.

How would you want pre-compilation to work? I didn't want to make them run `dustc`.

File: examples/streaming/app.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+var fs = require('fs'),
+    path = require('path'),
+    express = require('express'),
+    request = require('request'),
+    dust = require('dustjs-linkedin');
+
+dust.config.whitespace = true;
+
+dust.onLoad = function(tmpl, cb) {
+  fs.readFile(path.join('./views', tmpl + '.dust'), { encoding: 'utf8' }, cb);
```
Comment (jimmyhchan) [inline examples/streaming/app.js]: nit: this kinda scares me a little to allow a tmpl name to drive an fs.read without sanitizing it. I know this is all server side but the source is compiled and loadsourced ....

maybe escape/sanitize against `..` or pre-load the templates?

Comment (sethkinast) [inline examples/streaming/app.js]: Yeah, I get what you're saying-- I was just trying to keep the example simple.

I can refactor slightly to resolve the tmplname before joining it to `./views`, preventing directory traversal attacks via dynamic partial names.

Comment (sethkinast) [inline examples/streaming/app.js]: Updated and sanitized, thanks!

File: examples/basic-node/app.js (hunk 1 above)
Comment (jimmyhchan) [inline examples/basic-node/app.js]: you're right. CJS loading would have been nice.  There's `compileFn` and `renderSource` but both of those use `loadSource` underneath.

Comment (sethkinast) [inline examples/basic-node/app.js]: Cjs is on my roadmap yeah.

**Comment by jimmyhchan** [comment]
looks good thanks :+1:

**Pull request merged**